### DISCUSSION TITLE
Error when using fetchoptions in apolloclient

### DISCUSSION THREAD

**Discussion by paulrostorp** [Akryum/vue-apollo#514] [Q&A]
Hi,
I am running into an error when using fetchoptions in the apolloclient:
`Error sending the query 'QUERYNAME' ServerParseError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)`

When configuring the httplink like this :
`const link = new HttpLink({
  uri: 'http://localhost:4000',
  credentials: 'include',
  fetchOptions: {
    mode: 'no-cors'
  }
});`

Any advice ?

**Comment by hieucodecraft** [comment]
I had same the error.

**Comment by hieucodecraft** [comment]
> Hi,
> I am running into an error when using fetchoptions in the apolloclient:
> `Error sending the query 'QUERYNAME' ServerParseError: Unexpected end of JSON input at JSON.parse (<anonymous>)`
> 
> When configuring the httplink like this :
> `const link = new HttpLink({ uri: 'http://localhost:4000', credentials: 'include', fetchOptions: { mode: 'no-cors' } });`
> 
> Any advice ?

I solved it, it's belong to the stage of parse in the AST of GraphQL. I suggest should use the GraphiQL or GraphQL Playground to the queries or mutations etc before you run it in the code. Coz, the both of them can help you to detect the syntax. Good luck!
